src/Services/MappingService/MappingService.cs

```diff
@@ -73,6 +78,16 @@ public async Task<BookingRequest> MapBookingRequest(string sessionGuid, IElement
             };
         }
 
+        public void MapAppointmentId(AppointmentType appointmentType, FormAnswers answers)
+        {
+            Answers value = answers.Pages
+                .SelectMany(page => page.Answers)
+                .FirstOrDefault(answer => answer.QuestionId.Equals(appointmentType.AppointmentIdKey));
+
+            if (value is not null && value.Response is not null)
+                appointmentType.AppointmentId = new Guid((string)value.Response);
```

Comment:
should we throw an exception at this point if the booking was supposed to be mapped but no value was found in the answers?

Reply:
It wouldn't pass the IntegrityCheck unless it was using a Fake provider...  so if it's an empty guid and nothing is found it will remain an empty Guid, only when it's using a Fake provider. Could throw an error if the Guid is not valid...  but could also just add that into the integrity check to ensure the values in the option are a valid Guid?

Reply:
maybe i misread this piece of code, is this not resolving the AppointmentIdKey, if one has been specified we want to get its value in the answers in that case if somehow the questionID was entered wrong or the value is just not present we would want to throw an exception as that value is required to request the booing infomation.

Reply:
Removed null check... Previously the integrity checks already checked the form to check there was a corresponding QuestionId. I've improved that IntegrityCheck to check if exists and all the values are Guids. So now I've removed the null check as I know now it will exist, and will be a Guid.

Reply:
Also added that the source element for the booking appointment id must be on a previous page to the booking element in the Booking Form Integrity Check

Reply:
if more if the answer is missing at the time of request, so you build a form which uses a textbox for the appointmentId, you could build a journey which avoids that question with some branching logic, There isnt a way to validate this at build its when u are getting that answer IF it is missing/null we should throw an exception as its required, rather then requesting a appointment with an invalid id.